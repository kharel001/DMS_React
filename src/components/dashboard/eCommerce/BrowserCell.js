import React from 'react';
import Avatar from '@material-ui/core/Avatar';

const BrowserCell = ({browser}) => {
  const {id, image, title, subTitle} = browser;
  return (
    <div key={id} className="user-profile d-flex flex-row">
      <Avatar
        alt={title}
        src={image}
        className="user-avatar-sm mr-3"
        data-test="avatarComp"
      />
      <div className="user-detail">
        <h4 className="mb-0" data-test="titleComp">{title}</h4>
        <div className="user-description" data-test="subTitleComp">{subTitle}</div>
      </div>
    </div>

  );
};

export default BrowserCell;
