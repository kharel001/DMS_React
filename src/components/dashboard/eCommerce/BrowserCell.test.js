import React from 'react';
import BrowserCell from './BrowserCell';
import {
    configure,
    shallow,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})

const setup = (props = {}) => {
    return shallow( < BrowserCell {...props}  />
    )
}

// const findByTestAttr=(wrapper, val)=>{
//     return wrapper.find(`[data-test='${val}']`)
// }
let getContextFunction=jest.fn()
getContextFunction.mockReturnValue({
    stroke: "stroke",
});

const canvas={
    getContext:getContextFunction,

}
const dummy_props={
  browser:{
      id:"id1",
  image:"image1",
  title:"title1",
  subTitle: "subTitle1"
}}
describe("BrowserCell Comp",()=>{
it("Snapshot",()=>{
    const wrapper= setup(dummy_props);
    expect(wrapper).toMatchSnapshot()

})
it("Props",()=>{
    const wrapper= setup(dummy_props);
    // let avatarComp= findByTestAttr(wrapper,"avatarComp")
    let avatarComp = wrapper.find('[data-test="avatarComp"]')
    expect(avatarComp.prop('alt')).toEqual("title1")
    expect(avatarComp.prop('src')).toEqual("image1")
    let titleComp= wrapper.find('[data-test="titleComp"]')
    expect(titleComp.props().children).toEqual("title1")
    let subTitleComp= wrapper.find('[data-test="subTitleComp"]')
    expect(subTitleComp.props().children).toEqual("subTitle1")

})
})