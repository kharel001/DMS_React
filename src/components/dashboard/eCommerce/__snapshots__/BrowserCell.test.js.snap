// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowserCell Comp Snapshot 1`] = `ShallowWrapper {}`;
